TestProjects/ShaderGraph/Assets/CommonAssets/Editor/KeywordTests.cs

```diff
@@ -195,11 +195,7 @@ public void CanGetKeywordDeclarationForReals()
             var sb =  new ShaderStringBuilder();
             foreach (var keyword in m_Collector.keywords)
             {
-                string declaration = keyword.GetKeywordDeclarationString();
-                if(!string.IsNullOrEmpty(declaration))
-                {
-                    sb.AppendLine(declaration);
-                }
+                keyword.AppendKeywordDeclarationStrings(sb);
```

Comment:
This more optimal form just appends the keyword declaration string directly into the builder...  no temp string GC allocs